core/src/main/java/org/elasticsearch/rest/AbstractRestChannel.java

```diff
@@ -64,8 +68,15 @@ public XContentBuilder newBuilder(@Nullable BytesReference autoDetectSource, boo
             contentType = XContentType.JSON;
         }
 
-        String[] filters = useFiltering ? request.paramAsStringArrayOrEmptyIfAll("filter_path") :  null;
-        XContentBuilder builder = new XContentBuilder(XContentFactory.xContent(contentType), bytesOutput(), filters);
+        String[] includes = null;
+        String[] excludes = null;
+        if (useFiltering) {
+            Set<String> filters = Strings.splitStringByCommaToSet(request.param("filter_path", null));
+            includes = Strings.toStringArray(filters.stream().filter(f -> f.charAt(0) != '-').collect(toList()));
```

Comment:
you can also just use one predicate and invert it?

```
Predicate<String> include = f -> f.charAt(0) != '-';
Predicate<String> exclude = include.negate();
``` 

that seems safer to me?